common/src/main/java/com/wynntils/models/items/annotators/game/IngredientAnnotator.java

```diff
@@ -25,15 +25,16 @@ public ItemAnnotation getAnnotation(ItemStack itemStack, String name) {
 
         String ingredientName = matcher.group(1);
         String tierColor = matcher.group(2);
-        int tier = Models.IngredientProfiles.getTierFromColorCode(tierColor);
 
-        IngredientProfile ingredientProfile = Models.IngredientProfiles.getIngredient(ingredientName);
-        if (ingredientProfile == null) return null;
-        if (ingredientProfile.getTier().getTierInt() != tier) {
+        int tier = Models.Ingredient.getTierFromColorCode(tierColor);
+        IngredientInfo ingredientInfo = Models.Ingredient.getIngredientInfoFromName(ingredientName);
+        if (ingredientInfo == null) return null;
+
+        if (ingredientInfo.tier() != tier) {
```

Comment:
i'm always wary of "requiring" an item to have matching API data to be fully processed, and this is no exception. i would much prefer it if ingredients could still get basic functionality like inventory highlights even if we can't find a matching `IngredientInfo`

Reply:
I understand. But please note that this is not really a change compared to before; we required an IngredientProfile then, and an IngredientInfo now. It is the same thing but different name.

Handling "wynnitems" (gear and ingredients) that are not found in our database "as best we could" is certainly some kind of robustness improvement. But it also makes things more difficult, since then suddenly you don't know if your ingredient has an associated IngredientInfo or not, and need to handle that (probably very unlikely) case everywhere. And if you forget it, you get a NPE which is arguably worse. Or we have to juggle around Optionals everywhere.

I've been thinking of trying to make more elegant parsing of crafted items; possibly such code could spill over and maybe we could have like UnrecognizedGear and UnrecognizedIngredient? 

I hope you agree that we can take this code as it is, and continue this discussion afterwards. (I fully understand your concern, I'm just saying it's not clear to me how to do this so we don't pay a price that is higher than the gain.)